### DISCUSSION TITLE
Field alias broken since SQLModel 0.0.14

### DISCUSSION THREAD

**Discussion by DurandA** [fastapi/sqlmodel#725] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Optional

from pydantic import Field as PydanticField, BaseModel
from sqlmodel import Field, SQLModel


class PydanticHero(BaseModel):
    name: str
    secret_name: str = PydanticField(alias="secretName")


class SQLModelHero(SQLModel):
    name: str
    secret_name: str = Field(alias="secretName")


pydantic_hero = PydanticHero(name="Pydantic Company", secretName="Samuel Colvin")
sqlmodel_hero = SQLModelHero(name="Tiangolo", secretName="Sebastian Ramirez")
```

### Description

* Create a `Model` with a `Field` that contain an alias
* Using the alias to create the model throws a ValidationError
* This is inconsistent with Pydantic behavior and previous version of SQLModel (< 0.0.14)

### Operating System

Linux

### SQLModel Version

0.0.14

### Python Version

Python 3.12.0

### Additional Context

This issue appeared on SQLModel 0.0.14.

**Comment by feliche93** [comment]
@DurandA I was facing the same issue thanks for creating this!

**Comment by cycledriver** [comment]
Looks like the pydantic SchemaValidator.validate_python (https://docs.pydantic.dev/latest/api/pydantic_core/#pydantic_core._pydantic_core.SchemaValidator.validate_python) doesn't handle aliases.  That is what is used to set the fields of an object here:

https://github.com/tiangolo/sqlmodel/blob/0c7def88b5d9652bf9288738e2e9276d9dd24b5f/sqlmodel/_compat.py#L302

**Comment by DurandA** [comment]
related: https://github.com/tiangolo/sqlmodel/pull/774

**Comment by Oliversinn** [comment]
I can't make .model_dump(by_alias=True) work. It always uses the attribute name.

**Reply by samshipengs** [reply]
@Oliversinn did you end up making it work?

**Reply by Oliversinn** [reply]
Not really. I ended up doing a mapping function 💔

**Comment by marutron** [comment]
Reproduced on SQLModel 0.0.16. 
The solution proposed in #774 does not work.

upd: works by using #290

**Comment by perminovs** [comment]
Did anyone find a workaround? The one mentioned in #290 seems to be working only for table-models.

**Comment by benjamingorman** [comment]
Also facing this issue

**Comment by JonasHablitzel** [comment]
for me this workaround works for pydanticv 2.6 sqlmodel 0.0.14 its similar to what #774 does
```python
from sqlmodel import SQLModel, Field

class Test:
    def __init__(self, test):
        self.Test = test


class Item(SQLModel):
    test: str = Field(schema_extra={"validation_alias": "Test"})


t1= Test("1")


Item.model_validate(t1)
Item.model_validate({"Test": "1"})
```

**Reply by KunxiSun** [reply]
It works! Thanks

sqlmodel v0.0.18

**Reply by daffodilistic** [reply]
This also worked for me to rename a field to a custom name during serialization:

```python3
class AcmeCorpAsset(SQLModel):
    AssetId: str
    # Rename the field to AssignedUser as the asset can be transferred to a sister company
    AssignedUser: Optional['ExpensiveErpUser'] = Field(default=None, schema_extra={"validation_alias": "AcmeCorpUser"})
```

**Comment by DataOps7** [comment]
I'm facing this issue as well and the above solution doesn't help with dumping

**Comment by YuriiMotov** [comment]
Thanks for reporting!

Created issue to track this: https://github.com/fastapi/sqlmodel/issues/1536
